src/targets/Logary.Targets.Zipkin/Targets_Zipkin.fs

```diff
@@ -0,0 +1,141 @@
+
+    open Logary.Utils.Chiron.Mapping
+    open Logary.Utils.Chiron.Formatting
+
+    let private val2str = function
+        | String s -> s
+        | Float f -> f.ToString()
+        | Int64 i64 -> i64.ToString()
+        | BigInt big -> big.ToString()
+        | Fraction (x, y) -> x.ToString() + "/" + y.ToString()
+        | Object fields -> 
+            let json = Json.serialize fields
+            Json.format json
+        | Array vals -> 
+            let json = Json.serialize vals
+            Json.format json
+        | Bool b -> if b then "true" else "false"
+        | Binary (bin, contentType) -> contentType + "=" + Convert.ToBase64String bin
+        
+    let private val2ann orient timestampTicks = function 
+        | Gauge (v, u) -> Annotation(Units.formatWithUnit orient u v, DateTime timestampTicks)
+        | Derived (v, u) -> Annotation(Units.formatWithUnit orient u v, DateTime timestampTicks)
+        | Event t -> Annotation(t, DateTime timestampTicks)
+
+    let private val2bin (name:PointName, (Field (value, maybeUnit))) =
+        let key = name.ToString()
+        match value with
+        | Bool b -> Annotations.Binary(key, b)
+        | Binary (bin, contentType) -> BinaryAnnotation(key, bin, AnnotationType.Bytes)
+        | other -> Annotations.Binary(key, val2str other)
+
+    let private annotateSpan state (message : Message) = 
+        match Map.tryFind "spanId" message.context with
+        | Some (Int64 recId) ->
+            let spanId = (uint64 recId)
+            let span = 
+                match Map.tryFind spanId state.activeSpans with
+                | None -> makeSpan message
+                | Some s -> s
+            let annotation = val2ann Units.UnitOrientation.Suffix message.timestampTicks message.value
+            span.Record annotation
+            message.fields
+            |> Map.toSeq
+            |> Seq.map val2bin 
+            |> Seq.iter (span.Record)
+            { state with activeSpans = Map.add spanId span state.activeSpans }
+        | _ -> state 
+
+    let loop (config : ZipkinConf) (ri : RuntimeInfo) (requests : RingBuffer<TargetMessage>) (shutdown : Ch<IVar<unit>>) = 
+        let rec loop state : Job<unit> = 
+            let collector = state.collector
+            Alt.choose [ shutdown ^=> fun ack -> 
+                             match collector with
+                             | :? IDisposable as d -> d.Dispose()
+                             | _ -> ()
+                             (ack *<= () :> Job<_>)
+                         RingBuffer.take requests ^=> function 
+                         | Log(logMsg, ack) -> 
+                             job { 
+                                 let nextState = annotateSpan state logMsg
+                                 do! ack *<= ()
+                                 return! loop nextState
+                             }
+                         | Flush(ack, nack) -> 
```

Comment:
No that's not what it's designed to do. It's better to flush when you receive a message about EndRequest matching an active span.